ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java

```diff
@@ -220,6 +226,15 @@ public Result processTransaction(
     final WorldUpdater worldUpdater = worldState.updater();
     final MessageFrame initialFrame;
     final Deque<MessageFrame> messageFrameStack = new ArrayDeque<>();
+
+    // returnStack introduce on eip2315
+    final ReturnStack returnStack;
+    if (enableReturnStack) {
+      returnStack = new ReturnStack(MessageFrame.DEFAULT_MAX_RETURN_STACK_SIZE);
+    } else {
+      returnStack = new ReturnStack();
```

Comment:
What's the harm in just initializing the return stack with the proper size always? It would stop us from having to add this extra boolean parameter here.

Reply:
you're right this makes the code more simple and clean